tools/generate-force-kernels.py

```diff
@@ -0,0 +1,383 @@
+        terms[i] = terms[i].reshape(Nshift, -1, Nsample).transpose((1, 2, 0)).ravel()
+
+    (f_longrange, f_plummer, f_spline, f_erf,
+     p_longrange, p_plummer, p_spline, p_erf, ) = terms
+    print(len(terms[0]))
+
+    def stat(x, size):
+        x = x.reshape(-1, size)
+        mean = numpy.mean(x, axis=-1)
+        std = numpy.std(x, axis=-1)
+        return mean, std
+
+    rx, junk = stat(r, Nsample)
+    rp_1d, rp_1d_s = stat((p_spline - p_longrange) / p_spline, Nsample * Nshift)
+    rf_1d, rf_1d_s = stat((f_spline - f_longrange) / f_spline, Nsample * Nshift)
+    rp_erf, junk = stat((p_erf) / p_spline, Nsample * Nshift)
+    rf_erf, junk = stat((f_erf) / f_spline, Nsample * Nshift)
+
+    rp_1d[rx==0] = 1
+    rf_1d[rx==0] = 1
+    rp_erf[rx==0] = 1
+    rf_erf[rx==0] = 1
+
+
+    from matplotlib.figure import Figure
+    from matplotlib.backends.backend_agg import FigureCanvasAgg
+
+    figure = Figure(figsize=(8, 8))
+    figure.text(0.5, 0.91, "Split=%g" % Split, ha='center', va='center')
+    canvas = FigureCanvasAgg(figure)
+    ax = figure.add_subplot(221)
+    ax.set_title('potential')
+    l, = ax.plot(rx, rp_1d, '-', label='actual')
+    ax.fill_between(rx, rp_1d - 3 * rp_1d_s, rp_1d + 3 * rp_1d_s, color=l.get_color(), alpha=0.4, label='3-sigma')
+    ax.plot(rx, rp_erf, label='erf, spline', ls=':')
+    ax.legend()
+    ax = figure.add_subplot(222)
+    ax.set_title('force')
+    l, = ax.plot(rx, rf_1d, '-', label='actual')
+    ax.fill_between(rx, rf_1d - 3 * rf_1d_s, rf_1d + 3 * rf_1d_s, color=l.get_color(), alpha=0.4, label='3-sigma')
+    ax.plot(rx, rf_erf, label='erf, spline', ls=':')
+    ax.legend()
+
+    ax = figure.add_subplot(223)
+    ax.set_title('relative and zoom')
+    l, = ax.plot(rx, rp_1d, '-', label='actual, zero-crossing')
+    ax.fill_between(rx, -3 * rp_1d_s,  3 * rp_1d_s, color=l.get_color(), alpha=0.4, label='3-sigma')
+    ax.plot(rx, rp_erf / rp_1d - 1, label='erf, spline', ls=':')
+    ax.plot(rx, rp_erf, '--', label='erf, zero-crossing', color='gray')
+    ax.set_ylim(-0.02, 0.02)
+    ax.grid()
+    ax.legend()
+    ax = figure.add_subplot(224)
+    ax.set_title('relative and zoom')
+    l, = ax.plot(rx, rf_1d, '-', label='actual, zero-crossing')
+    ax.fill_between(rx,  - 3 * rf_1d_s,  3 * rf_1d_s, color=l.get_color(), alpha=0.4, label='3-sigma')
+    ax.plot(rx, rf_erf / rf_1d - 1, label='erf, spline', ls=':')
+    ax.plot(rx, rf_erf, '--', label='erf, zero-crossing', color='gray')
+    ax.set_ylim(-0.03, 0.03)
+    ax.grid()
+    ax.legend()
+
+    figure.savefig(ns.prefix + 'diagonstics-%.2f.png' % Split, dpi=200)
+
+
+    table = numpy.array([rx, rp_1d, rf_1d, rp_erf, rf_erf]).T
+
+    numpy.savetxt('shortrange-force-kernels.txt',
+        table,
+        header='x(in mesh units) w_pot_1d(x) w_force_1d(x) [erfc + other terms] w_pot_erf(x) w_force_erf(x) split=%.2f' % Split
+    )
+
+    # simply multiple these numbers to the unwindowed short-range force.
```

Comment:
Could you add more of a comment here, or a script that turns these output tables into the actual code that sits in gravity.c? Would help to ensure no confusion later.